### PR TITLE
Fix bug in selecting hdu indices in sunpy.io.fits.read

### LINKED ISSUE #4181: Example of reading data/header with io.fits.read is broken
In the docs, we have [the following example of how to read the data and header directly from a file using `sunpy.io.fits.read`](https://docs.sunpy.org/en/stable/code_ref/map.html#fixing-map-metadata),

```python
[data,header] = sunpy.io.fits.read(filename)
```

There are a few problems here.

1. If you have 2 HDUs, this returns the first to `data` and the second to `header`, obviously not the intended behavior.
2. You can use the `hdus` kwarg to select only a specific HDU but it turns out this is broken because the rest of the function expects an `HDUList` and after we slice it, we don't cast it back to an `HDUList` so everything after this explodes in fun and unexpected ways.
3. Even accounting for the first two problems, this function returns a list of data,header pairs, not a list of data and headers so you get a `ValueError` when trying to return this to a tuple as above. Fixing items 1 and 2, this example could be amended to,

```python
data, header = sunpy.io.fits.read(filename)[0]
```

### PR DESCRIPTION
**Pull request by wtbarnes**
Fixes #4181 and cleans up a few tests for `sunpy.io.fits`

### CODE REVIEW

File: sunpy/io/fits.py [hunk 1]
```diff
@@ -71,7 +71,8 @@ def read(filepath, hdus=None, memmap=None, **kwargs):
             elif isinstance(hdus, collections.Iterable):
                 hdulist = [hdulist[i] for i in hdus]
 
-        hdulist.verify('silentfix+warn')
+        hdulist = fits.hdu.HDUList(hdulist)
+        [h.verify('silentfix+warn') for h in hdulist]
```
Comment (ayshih) [inline sunpy/io/fits.py:75]: Is this acceptable coding style compared to an explicit loop or to a `map()` call?  It's a list comprehension to create a list that isn't actually used...  The answer can be "yes"; I'm simply asking.

Comment (wtbarnes) [reply sunpy/io/fits.py:75]: From the [standard library docs](https://docs.python.org/3/tutorial/datastructures.html#list-comprehensions):

> List comprehensions provide a concise way to create lists.

So no, this should probably just be a `for` loop.

**Comment by Cadair** [comment]
Can haz changelog?

File: sunpy/io/fits.py (hunk 1 above)
Comment (wtbarnes) [reply sunpy/io/fits.py]: Fixed!

**Comment by wtbarnes** [comment]
I added separate changelogs for the docfix and the bugfix. Do they need to be numbered as well? i.e. `4183.bugfix.1.rst` and `4183.docfix.2.rst`?

**Comment by ayshih** [comment]
Nope

File: changelog/4183.docfix.rst [hunk 2]
```diff
@@ -0,0 +1,2 @@
+Fixed example usage of `~sunpy.io.fits.read` to account for the fact that it returns a list
```
Comment (ayshih) [inline changelog/4183.docfix.rst:1]:
```suggestion
Fixed example usage of :func:`~sunpy.io.fits.read` to account for the fact that it returns a list
```

**Pull request merged**